app/src/main/java/org/mozilla/focus/download/DownloadInfoManager.java

```diff
@@ -92,7 +93,17 @@ protected void onQueryComplete(int token, Object cookie, Cursor cursor) {
                         List<DownloadInfo> downloadInfoList = new ArrayList<>();
                         if (cursor != null) {
                             while (cursor.moveToNext()) {
-                                downloadInfoList.add(cursorToDownloadInfo(cursor));
+                                DownloadInfo downloadInfo = cursorToDownloadInfo(cursor);
```

Comment:
We don't auto-delete items since items may be on SDcard and is removed temporarily only. Let's focus on showing error messages when trying to delete / access non exist items for this bug. See spec v2.0 Chapter 3. Download -Error Case (Cannot Find File).